client/src/templates/Challenges/components/scene/scene-helpers.test.ts

```diff
@@ -0,0 +1,341 @@
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: 'Hello world',
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe('\n<strong>Naomi</strong>: Hello world\n');
+    });
+
+    it('should build transcript from multiple commands with dialogue', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: 'Hello',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Quincy',
+          startTime: 1,
+          dialogue: {
+            text: 'Hi there',
+            align: 'right'
+          }
+        },
+        {
+          character: 'Naomi',
+          startTime: 2,
+          dialogue: {
+            text: 'How are you?',
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        '\n<strong>Naomi</strong>: Hello\n' +
+          '\n<strong>Quincy</strong>: Hi there\n' +
+          '\n<strong>Naomi</strong>: How are you?\n'
+      );
+    });
+
+    it('should skip commands without dialogue', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: 'Hello',
+            align: 'left'
+          }
+        },
+        {
+          character: 'Quincy',
+          startTime: 1
+          // No dialogue
+        },
+        {
+          character: 'Naomi',
+          startTime: 2,
+          dialogue: {
+            text: 'How are you?',
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        '\n<strong>Naomi</strong>: Hello\n' +
+          '\n<strong>Naomi</strong>: How are you?\n'
+      );
+    });
+
+    it('should handle dialogue with special characters', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: "I'm fine, thanks! How about you?",
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe(
+        "\n<strong>Naomi</strong>: I'm fine, thanks! How about you?\n"
+      );
+    });
+
+    it('should handle dialogue with punctuation', () => {
+      const commands: SceneCommand[] = [
+        {
+          character: 'Naomi',
+          startTime: 0,
+          dialogue: {
+            text: 'What? No way!',
+            align: 'left'
+          }
+        }
+      ];
+      const result = buildTranscript(commands);
+      expect(result).toBe('\n<strong>Naomi</strong>: What? No way!\n');
+    });
+
+    it('should handle dialogue with numbers', () => {
```

Comment:
We should fold this into either `should build transcript from single command with dialogue` or `should build transcript from multiple commands with dialogue`. Let's just have a dialogue text with a number in those cases.